src/dune_rules/lib.mli

```diff
@@ -100,8 +100,20 @@ module DB : sig
     val not_found : t
     val found : Lib_info.external_ -> t
     val to_dyn : t Dyn.builder
-    val redirect : db -> Loc.t * Lib_name.t -> t
+    val redirect : db -> Lib_id.t -> t
     val redirect_in_the_same_db : Loc.t * Lib_name.t -> t
+
+    module With_multiple_results : sig
+      type resolve_result := t
+
+      type t = private
+        | Resolve_result of resolve_result
+        | Multiple_results of resolve_result Nonempty_list.t
```

Comment:
Don't think this type gains us much over just using a list. The empty list can just correspond to `not_found` as well.